src/upgrd/upgrd.hxx

```diff
@@ -206,17 +206,18 @@ namespace upgrd {
                       !expected_sha1.empty()
                     ); 
                     
-                    auto system_shell = bp::shell();
+                    
 
                     _log << "We will close, the next time you will come back you will be in the newer version" << std::endl;
   
                     #if BOOST_OS_WINDOWS
-                    auto str_cmd = "timeout /t 3 /nobreak & del /F /Q "s + _app_path.make_preferred().string() + " & "
-                      + "move /Y " + upgraded_app.make_preferred().string() + " " + _app_path.make_preferred().string();
-
-                    bp::spawn(system_shell, "/c", str_cmd.data());
-
+                    _log <<"Tipi may need administrative privileges to apply the update."<< std::endl;
+                    auto system_shell = bp::search_path("powershell.exe");
+                    auto str_cmd = "Start-Process cmd -ArgumentList \'timeout /t 3 /nobreak & del /F /Q "s + _app_path.make_preferred().string() + " & "
+                      + "move /Y " + upgraded_app.make_preferred().string() + " "s + _app_path.make_preferred().string(); +"\' -Verb RunAs";
+                    bp::spawn(system_shell,"-Command",str_cmd.data());
```

Comment:
That should not compile IMHO. There's a semi-colon after a string() call that's in the way.  If it does for some reason I don't understand the result wouldn't be the expected one.

Have you actually tried it?

`_app_path.make_preferred().string(); +"\' -Verb RunAs";`
` -----------------------------------^`

Reply:
+ indenting and spaces. Btw. where's the style-guide you wanted to write?

Reply:
> Indeed it is a mistake on my part, I did not see it at all. The style-guide is currently being written and I'm doing it in the background

Regarding a style guide I have a 5 bullet points one ready that exists that I can push and will accomodate flexibility and readability. And by the way we have a clang-format autoformat settings in nxxm, which is also a description of the expected style.